app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,82 @@
+<% if can? :read , @recipe %>
+<div class="recipe-detail">
+<h1><%= @recipe.name %></h1>
+<div>
+<div class="detals">
+  <div class="info">
+  <div class="time">
+    <p class="fs-4">Preparation time:  </p><span class="fs-4"><%=@recipe.preparationTime%></span>
+    </div>
+    <div class="time">
+    <p class="fs-4">Cooking time:  </p><span class="fs-4"><%=@recipe.cookingTime%></span>
+    </div>
+  </div>
+  <% if can? :update, @recipe%>
+    <div class="status">
+      <%= form_with url: update_path(@recipe), method: "put" do |form| %>
+        <div class="form-check form-switch">
+          <% if @recipe.public == false %>
+            <%= form.check_box :public, class:"form-check-input fs-4", id:"flexSwitchCheckDefault", onchange: "this.form.submit();"%>
+            <label class="form-check-label fs-4" for="flexSwitchCheckDefault">Public</label>
+          <% else %>
+            <%= form.check_box :public, class:"form-check-input fs-4", id:"flexSwitchCheckChecked", checked: true, onchange: "this.form.submit();" %>
+            <label class="form-check-label fs-4" for="flexSwitchCheckChecked">Public</label>
+          <%end%>
+        </div>
+      <%end%>
+    </div>
+  <%end%>
+  </div>
+  <div>
+  <div class="descr bg-light mb-3">
+    <p class="fs-5"><%=@recipe.description%></p>
+  </div>
+  </div>
+  <div class="actions">
+    <%= render 'generatepopup' %>
+    <% if can? :update, @recipe%>

```

Comment:
- [ ] The app breaks and generates an error below when adding generating a list with empty fields. Kindly fix the errors and ensure that the app does not break.
![Screenshot from 2021-12-24 00-00-07](https://user-images.githubusercontent.com/20245164/147291451-6b1caa33-7d2f-44c5-a8f5-077483543fc5.png)